h/services/annotation_json_presentation.py

```diff
@@ -66,3 +67,25 @@ def annotation_json_presentation_service_factory(context, request):
                                              flag_count_svc=flag_count_svc,
                                              moderation_svc=moderation_svc,
                                              has_permission=request.has_permission)
+
+
+def annotation_jsonld_presentation_service_factory(context, request):
+    (group_svc, links_svc, flag_svc, flag_count_svc, moderation_svc) = _find_services(request)
+    return AnnotationJSONPresentationService(session=request.db,
+                                             user=request.user,
+                                             group_svc=group_svc,
+                                             links_svc=links_svc,
+                                             flag_svc=flag_svc,
+                                             flag_count_svc=flag_count_svc,
+                                             moderation_svc=moderation_svc,
+                                             has_permission=request.has_permission,
+                                             json_flavor='jsonld')
+
+
+def _find_services(request):
+    group_svc = request.find_service(IGroupService)
+    links_svc = request.find_service(name='links')
+    flag_svc = request.find_service(name='flag')
+    flag_count_svc = request.find_service(name='flag_count')
+    moderation_svc = request.find_service(name='annotation_moderation')
+    return (group_svc, links_svc, flag_svc, flag_count_svc, moderation_svc)
```

Comment:
This is not so much your fault, as the JSON presentation service really just relies on too many other services. But it could be made a bit nicer by returning a map instead of a tuple because then the ordering doesn't matter when we try to get one or some services out of it.